pandas/tests/plotting/test_hist_method.py

```diff
@@ -293,6 +293,28 @@ def test_hist_column_order_unchanged(self, column, expected):
 
         assert result == expected
 
+    @pytest.mark.slow
+    @pytest.mark.parametrize("by", [None, "b"])
+    @pytest.mark.parametrize("label", [None, "c"])
+    def test_hist_with_legend(self, by, label):
+        expected_labels = label or "a"
+        expected_axes_num = 1 if by is None else 2
+        expected_layout = (1, 1) if by is None else (1, 2)
+
+        index = 15 * [1] + 15 * [2]
+        s = Series(np.random.randn(30), index=index, name="a")
+        s.index.name = "b"
+
+        kwargs = {"legend": True, "by": by}
+        if label is not None:
+            # Behavior differs if kwargs contains "label": None
+            kwargs["label"] = label
```

Comment:
emm, why need to do this? is `label` an argument in `series/dataframe.hist`?

Reply:
Yes, it is passed as kwargs to the backend and seems relevant to test for it here. But perhaps the comment is not as clear as it should be. I think 

    # Don't add "label": None, causes different behavior than no label kwarg

may be better.

Reply:
yeah, then what's the default of `label`? isn't it `None`?

Reply:
There is no default. When not specified, the key "label" does not appear in kwargs. This is different than having "label" in kwargs and mapping to None. When "label": None is an entry in kwargs, you get warnings "No handles with labels found to put in legend." and no legend appears.

Reply:
okay, i see

Reply:
I took a third attempt at having a concise but clear comment to this regard:

    # We get warnings if kwargs contains "label": None
    if label is not None:

Hopefully that's better.

Reply:
thanks, this looks much clearer!